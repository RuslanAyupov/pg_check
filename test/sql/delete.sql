BEGIN;

CREATE EXTENSION pg_check;

-- no index
CREATE TABLE test_table (
    id      INT
);

INSERT INTO test_table SELECT i FROM generate_series(1,10000) s(i);

DELETE FROM test_table WHERE MOD(id, 2) = 0;

SELECT pg_check_table('test_table', true, true);

DROP TABLE test_table;

-- index
CREATE TABLE test_table (
    id      INT PRIMARY KEY
);

INSERT INTO test_table SELECT i FROM generate_series(1,10000) s(i);

DELETE FROM test_table WHERE MOD(id, 2) = 0;

SELECT pg_check_table('test_table', true, true);

DROP TABLE test_table;

ROLLBACK;
