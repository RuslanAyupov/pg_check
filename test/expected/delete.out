BEGIN;
CREATE EXTENSION pg_check;
-- no index
CREATE TABLE test_table (
    id      INT
);
INSERT INTO test_table SELECT i FROM generate_series(1,10000) s(i);
DELETE FROM test_table WHERE MOD(id, 2) = 0;
SELECT pg_check_table('test_table', true, true);
 pg_check_table 
----------------
              0
(1 row)

DROP TABLE test_table;
-- index
CREATE TABLE test_table (
    id      INT PRIMARY KEY
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "test_table_pkey" for table "test_table"
INSERT INTO test_table SELECT i FROM generate_series(1,10000) s(i);
DELETE FROM test_table WHERE MOD(id, 2) = 0;
SELECT pg_check_table('test_table', true, true);
NOTICE:  checking index: test_table_pkey
 pg_check_table 
----------------
              0
(1 row)

DROP TABLE test_table;
ROLLBACK;
